### DISCUSSION TITLE
Key error

### DISCUSSION THREAD

**Discussion by allenmoench** [earthlab-education/Earth-Analytics-AY24#582] [Q&A] [answered]
Getting a "key error". Anyone know how to resolve this? 
![image](https://github.com/user-attachments/assets/5be0c0f5-8d3b-442d-b8c2-2df71e9a9cf8)

**Comment by brglea** [comment]
Hi Allen! I'm going to include the code I used in this cell below. I think you need to change the end_index = -4 , and you could also try commenting out (using # in front of the later code section for opening the band and accumulate) and use .print(band_name) or path or id to make sure that's working before opening the band to accumulate.

Also I had to (multiple times) restart the kernel and rerun the code, so if fixing the code doesn't help I would do this and even close out of vs code entirely and reopen the app.

Hopefully some or all of this helps!

(Also the code copies funny into this cell so pay attention to your own indentation pattern.)

# Define band labels
bands = {
    'B01': 'aerosol',
    'B02': 'blue',
    'B03': 'green',
    'B04': 'red',
    'B05': 'nir',
    'B06': 'swir1',
    'B07': 'swir2',
    'B09': 'cirrus',
    'B10': 'thermal1',
    'B11': 'thermal2'
}

band_dict = {}
band_paths = glob(os.path.join(hls_data_dir, '*.B*.tif'))
for band_path in band_paths:
    print(band_path)

    # Get the band number and name
    start_index = -7
    end_index = -4
    band_id = band_path[start_index:end_index]
    print(band_id)
    band_name = bands[band_id]
    print(band_name)

    # Open the band 
    band_da = rxr.open_rasterio(band_path, mask_and_scale=True). squeeze()
    masked_da = band_da.where(cloud_mask)
    display(masked_da)

    masked_da.plot()

    # Accumulate
    band_dict[band_name] = masked_da
    
band_dict

**Reply by allenmoench** [reply]
Thanks!! I'll give this a try.

**Comment by eculler** [comment, marked as answer]
@allenmoench it looks to me like the problem here is that your dictionary keys are not matching what you are pulling out of the file name. And it looks to me like the reason for *that* is that you are missing the last character in the band name! Notice that it's looking for 'B0', rather than 'B0x'. 

In Python, when you index a range, it **does not include** the end of the range. That means you have to go one past what you might think. In this case, your end_index will need to be -4 in order to include the character at index -5.
